prow/test/integration/test/webhook_server_test.go

```diff
@@ -118,8 +126,151 @@ func TestWebhookServerMutateProwjob(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unable to unmarshal into prowjob %v", err)
 	}
+
 	diff := cmp.Diff(originalProwJob.Spec.DecorationConfig, mutatedProwJob.Spec.DecorationConfig)
 	if diff == "" {
 		t.Errorf("prowjob was not mutated %v", err)
 	}
 }
+
+func TestWebhookServerAvailability(t *testing.T) {
+	
+	clusterContext := getClusterContext()
+	kubeClient, err := NewClients("", clusterContext)
+	if err != nil {
+		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
+	}
+	ctx := context.Background()
+	t.Cleanup(func() {
+		var testProwjob v1.ProwJob
+		absDeploymentPath, err := filepath.Abs(deploymentPath)
+		if err != nil {
+			logrus.WithError(err).Info("could not find absolute path")
+		}
+		out, err := exec.Command("kubectl", "--context="+getClusterContext(), "apply", "-f", absDeploymentPath).CombinedOutput()
+		if err != nil {
+			logrus.WithError(err).Infof("could not clean up cluster %s", string(out))
+		}
+		key := types.NamespacedName{
+			Namespace: "default",
+			Name: testServerAvailability,
+		}
+		if err := kubeClient.Get(ctx, key, &testProwjob); err != nil {
+			logrus.WithError(err).Infof("prowjob not found")
+		} else {
+			out, err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", testServerAvailability).CombinedOutput()
```

Comment:
use kubeClient to delete

Reply:
like you said this is best effort delete, using kubeclient i would first have to get the object then pass the pointer into the delete. Is there need to do all that?

Reply:
yes, if possible we should avoid shelling out to execute `bash` command as much as possible, these are will only fail at runtime, there is no way to lint them and code review is much harder.

Reply:
done